Assets/LeapMotion/Core/Scripts/DataStructures/Pool.cs

```diff
@@ -85,7 +83,29 @@ public static class Pool<T> where T : new() {
       return value;
     }
 
+    private static T spawnValue() {
+      T value;
+      int count = 0;
+      do {
+        if (_pool.Count > 0) {
+          value = _pool.Pop();
+        }
+        else {
+          value = new T();
+        }
+        count++;
+      } while (value == null && count < 100);
+      if (count == 100) {
```

Comment:
It sounds like we should add null-checks to the recycle code then.  Wouldn't that solve the need to use a counter?